### PR TITLE
dotCMS/core#26057 Implement `chip` component with the new design system

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Implement `chip` component with the new design system

Screenshot 2023-09-15 at 12 36 56 AM

Screenshot 2023-09-15 at 12 37 11 AM

### CODE REVIEW

**Comment by zJaaal** [comment]
We have chips in autocomplete and multiselect. What about to unify the styles for all the components that uses chips? @fmontes

File: core-web/apps/dotcms-ui/src/stories/primeng/chip/templates.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export const ExampleVariations = `
+<div style="display: flex; gap: 24px; flex-direction: column; align-items: center">
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/stories/primeng/chip/templates.ts:2]: I am moving away from this a leaving the user use the controllers... check what I did in buttons... is not set in store, we might want to discuss.

File: core-web/libs/dotcms-scss/shared/_colors.scss [hunk 2]
```diff
@@ -104,8 +104,8 @@ $color-accessible-text-green-bg: $color-alert-green-light;
 $color-accessible-text-purple: $color-palette-secondary-600;
 $color-accessible-text-purple-bg: $color-palette-secondary-100;
 
-$color-accessible-text-blue: $color-palette-primary-500;
-$color-accessible-text-blue-bg: $color-palette-primary-200;
+$color-accessible-text-blue: $color-palette-primary-600;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/shared/_colors.scss:107]: What is is the diff between alert and this?

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss [hunk 3]
```diff
@@ -1,18 +1,82 @@
...
-        white-space: nowrap;
+        font-size: $font-size-md;
+    }
+
+    .p-chip-icon,
+    .pi-chip-remove-icon {
+        justify-content: center;
+        align-items: center;
+        display: flex;
+
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss:21]:
```suggestion
```

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss [hunk 4]
```diff
@@ -1,18 +1,82 @@
...
+        line-height: 1;
+        aspect-ratio: 1/1;
+        height: calc($icon-md + 4px);
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss:24]: Hummm... why?

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss [hunk 5]
```diff
@@ -1,18 +1,82 @@
...
+        font-size: $icon-md;
+    }
+
+    .pi-chip-remove-icon {
+        border-radius: $border-radius-sm;
+        transition: box-shadow 0.15s;
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss:30]: Why?

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss [hunk 6]
```diff
@@ -1,18 +1,82 @@
...
+    }
+
+    &.p-chip-image {
+        // Not implemented
+    }
+
+    &.p-chip-secondary {
+        background-color: $color-accessible-text-green-bg;
+        color: $color-accessible-text-green;
+    }
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss:44]: I don't think we need this.

File: core-web/libs/dotcms-scss/shared/_common.scss [hunk 7]
```diff
@@ -12,6 +12,10 @@ $field-height-lg: 3rem; // 48px
 
 $field-border-size: 1.5px;
 
+$chip-height-sm: 1.5rem; // 24px
+$chip-height-md: 2.5rem; // 40px
```
Comment (fmontes) [inline core-web/libs/dotcms-scss/shared/_common.scss:16]: Do this inside the chips if you actually need it.

**Review by fmontes — DISMISSED** [review]

File: core-web/libs/dotcms-scss/shared/_common.scss (hunk 7 above)
Comment (oidacra) [reply core-web/libs/dotcms-scss/shared/_common.scss:16]: Moved.

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss (hunk 4 above)
Comment (oidacra) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss:24]: `$icon-sm` is `14px` but in the figma is `16px` square. In the case of `$icon-md` is `16px` but the icon in the Figma is `20px`

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss (hunk 6 above)
Comment (oidacra) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss:44]: Fixed the vars.

File: core-web/libs/dotcms-scss/shared/_colors.scss (hunk 2 above)
Comment (oidacra) [reply core-web/libs/dotcms-scss/shared/_colors.scss:105]: Removed the alert one.

File: core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss (hunk 4 above)
Comment (oidacra) [reply core-web/libs/dotcms-scss/angular/dotcms-theme/components/_chip.scss]: handle the size of the icons globally

**Review by hmoreras — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/stories/primeng/chip/templates.ts (hunk 1 above)
Comment (oidacra) [reply core-web/apps/dotcms-ui/src/stories/primeng/chip/templates.ts]: File removed

**Review by manuelrojas — APPROVED** [review]

**Review by rjvelazco — APPROVED** [review]

**Review by zJaaal — APPROVED** [review]

**Pull request merged**